src/android/java/io/jxcore/node/BtConnectorHelper.java

```diff
@@ -68,119 +74,77 @@ public boolean isRunning(){
-                print_debug("Disconnect:::Stop :" + currentpeerId);
-                mBtToRequestSocket.Stop();
-                mBtToRequestSocket = null;
-                ret = true;
+        for (int i = 0; i < mServerSocketList.size(); i++) {
+            BtToServerSocket tmpToSrvSocket = mServerSocketList.get(i);
+            if (tmpToSrvSocket != null) {
+                print_debug("Disconnect:::Stop : mBtToServerSocket :" + tmpToSrvSocket.getName());
+                tmpToSrvSocket.Stop();
+                mServerSocketList.remove(i);
             }
         }
 
-        return ret;
-    }
+        mServerSocketList.clear();
 
-    //test time implementation to simulate 'peer disappearing'
-    public boolean  DisconnectIncomingConnections(){
-
-        boolean ret = false;
-        if (mServerSocketList != null) {
-            for(int i = 0; i < mServerSocketList.size(); i++) {
-                BtToServerSocket tmp = mServerSocketList.get(i);
-                if (tmp != null) {
-                    print_debug("Disconnect:::Stop : mBtToServerSocket :" + tmp.getName());
-                    tmp.Stop();
-                    tmp = null;
-                    mServerSocketList.set(i, tmp);
-                    ret = true;
-                }
-            }
+        return true;
 
-            mServerSocketList.clear();
-        }
-        return ret;
     }
 
     public String GetBluetoothAddress(){
-
-        String ret= "";
         BluetoothAdapter bluetooth = BluetoothAdapter.getDefaultAdapter();
-        if(bluetooth != null){
-            ret = bluetooth.getAddress();
-        }
-
-        return ret;
+        return bluetooth == null ? null : bluetooth.getAddress();
     }
 
     public String GetDeviceName(){
-
-        String ret= "";
         BluetoothAdapter bluetooth = BluetoothAdapter.getDefaultAdapter();
-        if(bluetooth != null){
-            ret = bluetooth.getName();
-        }
-
-        return ret;
+        return bluetooth == null ? null : bluetooth.getName();
     }
 
     public boolean SetDeviceName(String name){
-
-        boolean  ret= false;
         BluetoothAdapter bluetooth = BluetoothAdapter.getDefaultAdapter();
-        if(bluetooth != null){
-            ret = bluetooth.setName(name);
-        }
-
-        return ret;
-    }
-    public String GetKeyValue(String key) {
-        SharedPreferences sharedPref = jxcore.activity.getPreferences(Context.MODE_PRIVATE);
-        return sharedPref.getString(key, null);
-    }
-
-    public void SetKeyValue(String key, String value) {
-        SharedPreferences sharedPref = jxcore.activity.getPreferences(Context.MODE_PRIVATE);
-        SharedPreferences.Editor editor = sharedPref.edit();
-        editor.putString(key, value);
-        editor.commit();
-    }
-
-    public String  MakeGUID() {
-        return UUID.randomUUID().toString();
+        return bluetooth == null ? null : bluetooth.setName(name);
```

Comment:
I'm surprised this didn't cause a compiler error. This method returns a boolean but we are returning null? Null for a primitive type? In any case shouldn't this be `return bluetooth == null ? false : bluetooth.setName(name)`?